packages/components/code-connect/components/syn-optgroup.figma.ts

```diff
@@ -0,0 +1,45 @@
+import figma, { html } from '@figma/code-connect/html';
+import type SynOptgroup from '../../src/components/optgroup/optgroup.js';
+import type { ValidProperties } from '../core/types';
+
+// Synergy Web Component Connection
+figma.connect('https://www.figma.com/design/bZFqk9urD3NlghGUKrkKCR/Synergy-Digital-Design-System?node-id=30344-112405', {
+  example: ({
+    defaultSlot,
+    disabled,
+    prefixSlot,
+    style,
+    suffixSlot,
+  }) => html`
+    <syn-optgroup
+      disabled=${disabled}
+      style=${style}
+    >
+      <span slot="prefix">${prefixSlot}</span>
+      ${defaultSlot}
+      <span slot="suffix">${suffixSlot}</span>
+    </syn-optgroup>
```

Comment:
The code of the optgroup is not really correct. 
Normally in dev it would look like this: 
![image](https://github.com/user-attachments/assets/559f93b5-34ef-4946-92ed-cd5fcc614cea)

but it is shown as this in figma: 
![Bildschirmfoto 2025-07-08 um 08 31 09](https://github.com/user-attachments/assets/c0cbb744-9f26-4b6c-9607-0bb8cfda4b52)

The `defaultSlot` should be mapped to the `label` prop instead

```suggestion
    <syn-optgroup
      disabled=${disabled}
      style=${style}
      label= ${defaultSlot}
    >
      <span slot="prefix">${prefixSlot}</span>
      <span slot="suffix">${suffixSlot}</span>
    </syn-optgroup>
```

Reply:
Ah, now I get it :). The problem with syn-optgoup is twofold:

1. There is now way to map a string, to boolean, so I would have to use `<span slot="label">${defaultSlot}</span>` in there. I now opted to use the `label` property like suggested.
2. The component does not take children as our `<syn-optgroup>` does in code. This means the output will always be incorrect when it is used. There is no way to fix this other than recreate the component in figma from scratch to allow a list of children that we can render. I created a todo comment for this.